core/src/main/java/org/apache/calcite/plan/RelOptMaterializations.java

```diff
@@ -214,7 +232,11 @@
     hepPlanner.setRoot(root);
     root = hepPlanner.findBestExp();
 
-    return new SubstitutionVisitor(target, root).go(materialization.tableRel);
+    return new SubstitutionVisitor(target, root, ImmutableList.
+        <SubstitutionVisitor.UnifyRule>builder()
+        .addAll(materializationRules)
+        .addAll(SubstitutionVisitor.DEFAULT_RULES)
```

Comment:
So, Real running rules of unify is collection of SubstitutionVisitor's default and external-injected?
Will we meet one case, which it only want work with external-injected rules, not included default?
This case may occur, for example: we overload some `SqlAggFunction` to replace calcite default `SqlAggFunction`.

Reply:
Your example is specialized and does not apply to `Calcite`.
In `Calcite`, it should be extended on the default rules.
If you have special needs, external rules can also be registered.

Reply:
This seems to be a new change. As @wojustme pointed out, this seems to prevent anyone from using their own materialization rules without including the default ones. Can we remove this line? Same result should be obtained by passing the default rules as part of the externally injected ones.